source/MaterialXGenEssl/CMakeLists.txt

```diff
@@ -0,0 +1,57 @@
+file(GLOB_RECURSE materialx_source "${CMAKE_CURRENT_SOURCE_DIR}/*.cpp")
+file(GLOB_RECURSE materialx_header "${CMAKE_CURRENT_SOURCE_DIR}/*.h")
+
+function(assign_source_group prefix)
+    foreach(_source IN ITEMS ${ARGN})
+        if (IS_ABSOLUTE "${_source}")
+            file(RELATIVE_PATH _source_rel "${CMAKE_CURRENT_SOURCE_DIR}" "${_source}")
+        else()
+            set(_source_rel "${_source}")
+        endif()
+        get_filename_component(_source_path "${_source_rel}" PATH)
+        string(REPLACE "/" "\\" _source_path_msvc "${_source_path}")
+        source_group("${prefix}\\${_source_path_msvc}" FILES "${_source}")
+    endforeach()
+endfunction(assign_source_group)
+
+assign_source_group("Header Files" ${materialx_header})
+assign_source_group("Source Files" ${materialx_source})
+
+add_library(MaterialXGenEssl STATIC 
```

Comment:
I've discussed this with @bernardkwok off-line, and my recommendation would be to add the new ESSL generator and syntax classes to the existing MaterialXGenGlsl library.  This would allow all GLSL generation functionality to be accessed from a single C++/Python library, and would help to focus developers on the more central distinction between GLSL, OSL, and MDL generators.